### ISSUE TITLE
Palindrome Bug: 0_0 (: /-\ :) 0-0 should NOT be true. Test is incorrect.

### ISSUE THREAD

**Issue by ajsnetiker** [FreeCodeCamp/FreeCodeCamp#6058]
Challenge [Bonfire: Check for Palindromes](http://www.freecodecamp.com/challenges/bonfire-check-for-palindromes#?solution=function%20palindrome(str)%20%7B%0A%20%20%2F%2F%20Good%20luck!%0A%20%20var%20checkArray%20%3D%20%5B%5D%3B%0A%20%20var%20word%20%3D%20str.toLowerCase()%3B%0A%20%20var%20newWord%20%3D%20word.replace(%2F%5B%2C%20.%5D%2Fg%2C%20%22%22)%3B%20%0A%20%20var%20reversed%20%3D%20newWord.split(%27%27).reverse().join(%27%27)%3B%0A%20%20console.log(%22Forward%3A%20%22%20%2B%20newWord)%3B%0A%20%20console.log(%22Backwards%3A%20%22%20%2B%20reversed)%3B%0A%20%20if(newWord%20%3D%3D%3D%20reversed)%20%7B%0A%20%20%20%20return%20true%3B%0A%20%20%7D%20%0A%20%20%20%20return%20false%3B%0A%20%20%7D%0A%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:43.0) Gecko/20100101 Firefox/43.0</code>.

There's a clear bug with the very last Palindrome, as visually it's clear to see 0_0 (: /-\ :) 0-0 should NOT return true. The beginning shows "0_0 (", the end is ")0-0" while Visually, these appear SYMMETRICAL but it's not a palindrome.

Attached are two screenshots. The first one is the results of my test, and second is everything displayed in the console. I made it easy by identifying each string forward and backwards and you can see the last one clearly doesn't match, as I stated above.
![palindromeconsole](https://cloud.githubusercontent.com/assets/13673808/12244485/c1f8733e-b858-11e5-9a45-4c6bf61f83da.jpg)
![palindrometest](https://cloud.githubusercontent.com/assets/13673808/12244486/c1fda76e-b858-11e5-9f01-9dd7efd5b4e9.jpg)

My code:
```javascript
function palindrome(str) {
  // Good luck!
  var checkArray = [];
  var word = str.toLowerCase();
  var newWord = word.replace(/[, .]/g, ""); 
  var reversed = newWord.split('').reverse().join('');
  console.log("Forward: " + newWord);
  console.log("Backwards: " + reversed);
  if(newWord === reversed) {
    return true;
  } 
    return false;
  }


```

**Comment by SaintPeter** [comment]
Nope, you're wrong.  The test is correct.

The instructions are to remove all non-alphanumeric characters, IE: anything which is NOT 0-9 and a-z.  If you do so, you'll end up with `0000`, which is a palindrome, for the purposes of the exercise.

You should revisit your regex, which is not nearly expansive enough.  I suggest http://regexr.com for reference and testing.

Finally - I encourage you to take your problems to the [HelpBonfires](https://gitter.im/FreeCodeCamp/HelpBonfires) channel where you will get excellent support by knowledgeable people who would have much more quickly told you the same thing.

Regards and Happy Coding!

**Comment by ajsnetiker** [comment]
Thanks! I must have missed the part about deleting ALL non-alphanumeric characters. I got it correct now. I still think the question is somewhat misleading though, you technically COULD check with characters in it and you would say "is this string the same backwards as it is forwards?" and the answer would have been NO for the last one.

It seems like cheating to remove symbols THEN check. My impression was that a palindrome was what I just described, the same in both directions, WITHOUT removing anything.

Weird, but I understand your point too, it's a matter of following the instructions.
